Dockerfile

```diff
@@ -2,19 +2,23 @@ FROM python:3.6-slim-stretch
 
 MAINTAINER Clifton Barnes <clifton.barnes@rovercode.com>
 
-RUN apt-get update
-RUN apt-get install -y build-essential python3-dev libi2c-dev python3-smbus
-
-ADD pytest.ini /var/rovercode/pytest.ini
-ADD requirements.txt /var/rovercode/requirements.txt
+ADD pytest.ini requirements.txt /var/rovercode/
 ADD GrovePi /var/rovercode/GrovePi
```

Comment:
You could shed a layer here by combining 

```
ADD GrovePi /var/rovercode/GrovePi
ADD rovercode /var/rovercode/rovercode
```

To 
```
COPY GrovePi rovercode /var/rovercode/
```

Reply:
My understanding was that if they are directories then it only copies the contents:
https://stackoverflow.com/questions/37715224/copy-multiple-directories-with-one-command#37715522